hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java

```diff
@@ -48,29 +46,44 @@
 
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
-	private final TokenStore store;
+	private final TokenStore commonStore;
+	private final TokenStore uniqueStore;
 	private final TransactionContext txnCtx;
 
+	// Todo add the 2 types of stores (their interfaces) here
 	public TokenMintTransitionLogic(
-			TokenStore store,
+			CommonStore commonStore,
+			UniqueStore uniqueStore,
 			TransactionContext txnCtx
 	) {
-		this.store = store;
+		this.commonStore = commonStore;
+		this.uniqueStore = uniqueStore;
 		this.txnCtx = txnCtx;
 	}
 
+	// TODO temporary fix
+	public TokenMintTransitionLogic(
```

Comment:
Q: Why don't we use the other constructor?

Reply:
Because the current InvertibleHashMap implementation is not a descendant of MerkleNode, and thus cannot  be loaded in ServicesContext